### PR TITLE
build(webpack): polyfill `setImmediate`

### PR DESCRIPTION
**Pull request by davidmurdoch**
Fixes flaky test `Sentry errors after initialization, after opting into metrics @no-mmi should send error events in background`  in the webpack build.

### CODE REVIEW

**Comment by HowardBraham** [comment]
LGTM, I ran the `Sentry errors after initialization, after opting into metrics @no-mmi should send error events in background` test 500 times in `test-e2e-chrome-webpack`, and it passed all 500 times!

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/95470/workflows/a2831e03-34dc-4964-b804-3612741b95a9/jobs/3553634/steps

File: development/webpack/webpack.config.ts [hunk 1]
```diff
@@ -256,7 +264,7 @@ const config = {
         test: /\.(?:js|mjs)$/u,
         include: NODE_MODULES_RE,
         // never process `@lavamoat/snow/**.*`
-        exclude: /^.*\/node_modules\/@lavamoat\/snow\/.*$/u,
+        exclude: [/^.*\/node_modules\/@lavamoat\/snow\/.*$/u],
```
Comment (davidmurdoch) [inline development/webpack/webpack.config.ts:267]:
```suggestion
        exclude: /^.*\/node_modules\/@lavamoat\/snow\/.*$/u,
```

File: package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "mv3:stats:chrome": "SELENIUM_BROWSER=chrome ts-node test/e2e/mv3-perf-stats/index.js",
     "user-actions-benchmark:chrome": "SELENIUM_BROWSER=chrome ts-node test/e2e/user-actions-benchmark.js",
     "benchmark:firefox": "SELENIUM_BROWSER=firefox ts-node test/e2e/benchmark.js",
-    "build:test": "yarn env:e2e build test",
+    "build:test": "yarn env:e2e build test --apply-lavamoat=false",
```
Comment (davidmurdoch) [inline package.json:34]:
```suggestion
    "build:test": "yarn env:e2e build test",
```

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by HowardBraham — APPROVED** [review]

**Pull request merged**